lib/resources/forms.js

```diff
@@ -109,7 +109,8 @@ module.exports = (service, endpoint) => {
         .then((partial) => Promise.all([
           Forms.createVersion(partial, form, false),
           Submissions.clearDraftSubmissions(form.id)
-        ])))
+        ]))
+        .then(() => Forms.purgeUnneededDrafts(form))) // remove drafts made obsolete by new draft
```

Comment:
naming: maybe call this clear or hardDelete or something other than the other kind of purging on a cycle that we do